src/ContractCallDataEncoder.cpp

```diff
@@ -361,7 +361,7 @@ QVariant ContractCallDataEncoder::decode(SolidityType const& _type, bytes const&
 	else if (type == QSolidityType::Type::Bool)
 		return QVariant::fromValue(toString(decodeBool(rawParam)));
 	else if (type == QSolidityType::Type::Bytes || type == QSolidityType::Type::Hash)
-		return QVariant::fromValue(toChar(decodeBytes(rawParam)));
+		return QVariant::fromValue(QString::fromStdString(toHex(decodeBytes(rawParam), 2, HexPrefix::Add)));
 	else if (type == QSolidityType::Type::String)
 		return QVariant::fromValue(toChar(decodeBytes(rawParam)));
```

Comment:
Can you explain a bit more, please? `bytes` and `string` are identical as far as the ABI is concerned. How does `ContractCallDataEncoder::decode` behave for bytes and for strings and how should it behave?